### DISCUSSION TITLE
UCB CS61C: Great Ideas in Computer Architecture - CS自学指南

### DISCUSSION THREAD

**Discussion by giscus** [PKUFlyingPig/cs-self-learning#80] [Announcements]
# UCB CS61C: Great Ideas in Computer Architecture - CS自学指南

CS自学指南

https://csdiy.wiki/%E4%BD%93%E7%B3%BB%E7%BB%93%E6%9E%84/CS61C/

**Comment by oneMelone** [comment]
project的初始代码是在哪里找的呀0-0

**Reply by ZephyrZenn** [reply]
https://github.com/61c-teach

**Comment by tuchao1996** [comment]
mark

**Comment by cwisspecial** [comment]
现在作业要登入一个啥账号才能做，有大佬能解决吗？

**Reply by trace1729** [reply]
作业没法写，lab 和 proj 足够了

**Comment by Somoku** [comment]
可能有人会想知道CS61C和CMU 15-213的不同之处，在此讲述一下我的个人体验：

- CS61C涵盖了CSAPP这本书的前六章和第九章的主题，只是在内容上有所取舍（例如对流水线hazard的讲述并没有CSAPP那样详细），并且更倾向于从硬件设计者的角度讲述，会涉及到底层逻辑电路的实现；而CMU 15-213是从程序员的角度讲述计算机系统，除了体系结构的知识外还包括很多操作系统的内容（课程后半部分），毕竟一门课是*Great Ideas in Computer Architecture*，另一门课是*A Programmer's Perspective*.
- 在汇编语言方面，CMU 15-213讲的是x86-64指令集，而CS61C讲的是RSIC-V指令集，在学习过程中可以感受到RISC和CISC的特点和区别，而不是局限于CSAPP里的那个表格。
- 在并行方面，个人感觉CS61C对SIMD指令的讲解很好地填补了我在学CMU 15-213时对这方面的模糊理解，CSAPP里这部分内容给我留下的印象就是繁杂的指令，让人摸不着头脑。
- 对我而言，CS61C最有价值的就是project2和project3了。project2要求自己写汇编代码，很好地帮助我熟悉RISC-V汇编指令和加深了我对调用者/被调用者保护的理解；project3要求自己设计一个二阶流水线CPU，这对理解data path和control logic的硬件实现非常有意义。并且，这两个project都有要求自己写测试样例，这是我未曾体验过的。相比之下，CMU 15-213的bomblab更考验对汇编指令的熟悉程度和GDB的使用，archlab则是基于对流水线的理解去优化代码。
- 当然，CS61C也有介绍一些CMU 15-213没有涉猎的，例如Cache coherence, MapReduce, RAID等。

总而言之，对于学过CMU 15-213的人而言，CS61C更像是对CSAPP前半本书的回顾，从硬件设计者的角度更好地理解计算机体系结构；对于没学过的人而言，如果精力有限，CS61C也不失为一门好的计算机体系结构入门课。当然，如果想对计算机整体有更深入的了解，依然强烈推荐阅读CSAPP这本大部头。

**Reply by ShaneHeX** [reply]
学过 15 年的 cs61c，感觉没 20 版的好

**Reply by Unik-lif** [reply]
各位要考虑到csapp的难度，cs61c对于像我这样的新手来说很舒服，但csapp的流程还是比较陡峭的，要花不少时间。

**Reply by lsy1109** [reply]
61c和国内同样使用软硬件接口的机组课程有什么区别呢

**Reply by cwisspecial** [reply]
cs61c\su20\lab07是关于缓存的实验，有没有大佬做了的，能教一下不？
没有一点思路，呜呜呜

**Reply by bing1117** [reply]
大佬推荐先学哪一个呢

**Reply by aka556** [reply]
同问

**Comment by ghost** [comment]
61C的playlist，质量非常高（我认为比录播好多了），并且很新
https://inst.eecs.berkeley.edu/~cs61c/fa20/#lectures

**Reply by Unik-lif** [reply]
Dan Garcia讲的好，另一个老师英语不太好（可以平替推荐sp22版本）

**Reply by soondubu137** [reply]
这个lecture videos确实很好！就是Nikolic教授的录音质量一般，听着有点费劲...

**Reply by exile0627** [reply]
同问sp22版本的lecture哪里可以看呢

**Reply by ajwwja777** [reply]
fall20的所有lec在ytb上都有
https://www.youtube.com/watch?v=VJ6tuX5bBf4&list=PL0j-r-omG7i0-mnsxN5T4UcVS1Di0isqf

**Reply by sleepy-shawn** [reply]
我倒是觉得bora比dan讲得好 dan讲的太快了 一直要复习ppt bora老师讲的刚刚好 本身也是技术大牛 很透彻

**Reply by Zlatanwic** [reply]
已经需要ucb的帐号了请问大佬怎么办呢

**Comment by richen666** [comment]
有获取homework的大佬，可以分享一下么？非常想练习一下这个hw

**Reply by EricZheng0404** [reply]
Google一下就有

**Reply by calyle** [reply]
搞到homework了吗

**Comment by vcerfe** [comment]
可以不听视频直接看PPT学习吗？（内容上会不会有缺失，英语听力不太好

**Reply by ShaneHeX** [reply]
会有缺失，而且老师会有不一样的思路

**Reply by HangX-Ma** [reply]
个人感觉 ’流水线‘ 那一部分需要看视频，思路会清晰，ppt在那部分就是流程式的，没有视频搞不明白的。

**Comment by Derick317** [comment]
有无朋友做过 cs61c project 1 `Philphix`? 大家用什么方法判断输入结束的呢? 我用 `EOF` 判断, 但那个二进制文件似乎遇到 `EOF` 也没有结束. (`EOF` 应该等于 -1)

**Reply by unuing** [reply]
那个project我[用fread读取stdin](https://github.com/unuing/CS61C-SU21-proj1/blob/c38b9a404387462f29d282dc697d895b0ab40b33/src/philphix.c#L188)就能过二进制那个test（fread判断文件结束是看返回值是不是等于第三个参数），但[getc就不行](https://github.com/unuing/CS61C-SU21-proj1/blob/afb30651a6da8a56c0ab0075a1e136b3a5a3361f/src/philphix.c#L206)，我也不知道为什么（我做的是summer 2021的）

**Comment by hbinfrog** [comment]
有没有小伙伴一起学习的

**Reply by dengyunsheng250** [reply]
有，可以建一个交流群不？

**Reply by Touger** [reply]
建群+1

**Reply by mark456tung** [reply]
+1

**Reply by xydetoli** [reply]
蹲

**Reply by dengyunsheng250** [reply]
建了一个群，cs61c交流群134367194

**Reply by jiangruixin** [reply]
群搜不到哎

**Reply by lizhi317** [reply]
插眼，同没找到群，有无好兄弟拉个群讨论一下下

**Reply by Owenun** [reply]
找不到上面的群，拉了一个新群，群号 769319660

**Comment by colinwgg** [comment]
请问推荐哪个版本呢？

**Reply by unuing** [reply]
可以看fall20的

**Reply by unuing** [reply]
也可以做summer21的

**Reply by xingyuwu821** [reply]
请问20fall和21fall哪个好？20fall老师上课比较有激情但是课件是黑底的不方便做笔记并且课程时长超过44h。21fall各方面都还行但是老师上课有点有气无力。

**Comment by jamesyaojiaxu** [comment]
我认为cs61c最好的版本是fall 2020
观看地址：https://www.bilibili.com/video/BV1jy4y1E7AY/
ppt下载地址：https://pan.baidu.com/s/1HdK0ddx885UfuoJIKlPFCw?pwd=1hpn 
官网：https://cs61c.org/fa20/

**Reply by serene-ding** [reply]
作业 lab 和 project 可以做吗

**Reply by jamesyaojiaxu** [reply]
可以，质量很好

**Reply by xingyuwu821** [reply]
请问20fall和21fall哪个好？20fall老师上课比较有激情但是课件是黑底的不方便做笔记并且课程时长超过44h。21fall各方面都还行但是老师上课有点有气无力。

**Reply by jamesyaojiaxu** [reply]
个人认为20fall好一点
1.黑底的问题,官网部分ppt提供白底,点击[ Slides Light ]或[ Slides BW ]下载即可.
2.时长问题.个人认为Bora Nikolic没有Dan Garcia讲的精彩.它的课可以有所取舍,参考Computer Organization and Design The Hardware Software Interface [RISC-V Edition] (David A. Patterson, John L. Hennessy)就行

**Reply by xingyuwu821** [reply]
好的，多谢你的解答。另外，我想问一下20fall和21fall的授课内容有很大差距吗？视频时长似乎差别很大，然后在官网上也可以看到20fall有40个lecture但是21fall只有27个lecture。我还发现对于其他的semester，有的时长是40个lecture，有的是27个lecture，很奇怪。

**Reply by jamesyaojiaxu** [reply]
21fall的学时被压缩了,从一周3节变为一周2节.它们的总体内容差不多,只是20fall讲的更细更精

**Reply by xingyuwu821** [reply]
我明白了，多谢，我觉得还是细点好，20fall更好。

**Reply by jamesyaojiaxu** [reply]
不客气

**Reply by zzqDeco** [reply]
我想问一下这个gradescope code如何获得

**Reply by zzqDeco** [reply]
20fall的open程度不敢恭维啊，hw全都不open，proj的视频也不open，太难受了

**Comment by Strive236** [comment]
请问这门课的gradescope code在哪找的呀，看lab0setup里面并没有

**Reply by trace1729** [reply]
没有gradescope，只能本地测试，也足够了

**Comment by calyle** [comment]
家人们，谁有cs61c和homework啊？

**Reply by zhangbird050801** [reply]
类似于 github 上 https://github.com/Junren0716/CS61c.git 每个文件夹种有各自 的Spec.pdf 描述了作业的步骤和要求

**Reply by exile0627** [reply]
目前课程官网能打开homework链接不需要登录的好像都在spring 2015之前了 https://inst.eecs.berkeley.edu/~cs61c/sp15/

**Comment by ghost** [comment]
学国外的课程需要看discussion吗

**Reply by DanielDaniel2201** [reply]
我正在cs61c，觉得挺有必要的，因为除了discussion自己输出知识的机会就只有lab和project了，但是这就比学生少了很多锻炼机会了，只是被动的看课和ppt不多运用感觉我的学习质量还是堪忧，而且discussion理论知识更多，有时候自己以为懂了，一进去做题就抓瞎，so感觉还是可以看看的

**Reply by exile0627** [reply]
请问cs61c各学期discussion notes的区别大嘛，有没有哪个学期discussion video质量比较好呢

**Comment by Lanisera** [comment]
有做过的朋友知道没hive机器对project4有影响吗

**Reply by QWenQ** [reply]
同学，你p4做了吗？没hive有什么影响啊？

**Reply by Lanisera** [reply]
还没有，proj3刚写完

**Reply by Knight-of-Thunder** [reply]
同学，有结论了吗，有佬做完24fa的说不能在本地做proj4

**Comment by DreamancerYxy** [comment]
萌新想问问61c-teach仓库里面那个GradescopeBase有人知道怎么用吗？？或者说project和lab要怎么在本地测试呢？

**Reply by DanielDaniel2201** [reply]
project lab的页面里都会写用什么命令本地测试，lab和project文件夹都有本地测试文件

**Comment by jiangruixin** [comment]
有没有同学学习的，加我微信Easylovesniperj  一起交流，加的人多可以拉个群

**Comment by Lanisera** [comment]
又朋友做了proj3测试代码的时候会很慢吗，我测part_b测了10分钟左右

**Comment by chiarua** [comment]
写汇编的时候，我发现我用不了venus，猜测是磁盘文件系统的问题（它要VMFS似乎）
有用windows能做的好哥哥请回复下

**Reply by Eleven-6** [reply]
做 20 fall 的 lab3， 如果在 Venus 的终端中显示 mount: IllegalStateException: This version of Venus's drive handler ([object Object]) is not compatible with the mounted file server (1.0.0)!， 原因是 tools 中的 venus.jar 不是最新的，替换为使用 23 fall tools/download_tools.sh 下载的 venus.jar 就可以正常使用 venus.

**Comment by Eleven-6** [comment]
做 20 fall 的 lab3， 如果在 Venus 的终端中显示 mount: IllegalStateException: This version of Venus's drive handler ([object Object]) is not compatible with the mounted file server (1.0.0)!， 原因是 tools 中的 venus.jar 不是最新的，替换为使用 23 fall tools/download_tools.sh 下载的 venus.jar 就可以正常使用 venus.

**Reply by linzhenpl07** [reply]
请问下有lab的任务说明吗？

**Comment by Owenun** [comment]
2024开始学习的伙伴们加个群一起讨论呗，群号 769319660

**Reply by Wang-Jasper** [reply]
没搜到啊

**Comment by yaoyhu** [comment]
前面很多群号都搜不到，于是建了个交流组：https://t.me/+KzTpltz33VRlOTA1

**Comment by dghtucs** [comment]
建了一个QQ群，讨论可入：729044101

**Comment by FSReed** [comment]
刚完成20 Fall project4的Task3， 个人感觉与Parallelism无关的准备工作真的相当多，Task1-3尽早开始是没错的，尤其是像我一样没接触过Python-C API的同学们，应该需要大量阅读官方文档
另外并没有必要用Python3.6版本，用3.6是因为要用到dumbpy库来测试自己的代码实现，而这个库只有在HIVE machine上才有，所以用别的Python版本，自己编写一些测试样例就可以（我用的Python3.8.19）

**Reply by smokingsoda** [reply]
同学请问一下如果在make test里出现了clang: error: unsupported option '-fopenmp'应该怎么解决呀 我是macbook

**Comment by Eleven-6** [comment]
在做 23fall-proj2 时如果出现了，“UnicodeDecodeError: 'gbk' codec can't decode byte 0xae in position 25: illegal multibyte sequence” 这样的的错误，可以在命令行中 export PYTHONUTF8=1 解决

**Comment by exile0627** [comment]
请问最新24 spring的lab和project是都要hive machine才能做了吗

**Comment by CrispyCandy750** [comment]
fa20和su20页面好像g了，现在要登录了，请问有大佬保存了原来的页面了吗

**Reply by CrispyCandy750** [reply]
这里我使用了Wayback Machine可以回溯到原先的页面版本

**Reply by zhangbird050801** [reply]
我昨天还可以看的，今天确实变成要登录了

**Reply by EdogawaAi** [reply]
果然有人提这个issue了。昨天fa20课程网站都能看，今天的话一上网啥都看不了😢2024sp的还能随便看lab和proj要求

**Reply by Eleven-6** [reply]
24 sp 和 23 fall 的 project 是一样的吗,（感觉差不多），23 fall 的 project3 做到一半😂

**Reply by missanchor** [reply]
https://web.archive.org/web/20240423100311/https://inst.eecs.berkeley.edu/~cs61c/fa20/

**Comment by missanchor** [comment]
cs61c网站有些要账号才能登了，我用Wayback Machine找了一个历史版本的，大家可以用这个https://web.archive.org/web/20240423100311/https://inst.eecs.berkeley.edu/~cs61c/fa20/

**Reply by 123587** [reply]
proj里图片显示不了

**Reply by killus-tianyi** [reply]
你好，请问你找到了能够显示的版本吗？

**Reply by Xuecmbm** [reply]
lab4打不开  请问您还有备份吗

**Comment by wozonet1** [comment]
现在好像必须是注册学生才能看过去课程了,否则校外的只能看到目前最新的课程(也就是CS61C sp2024),,虽然能找到过去的lab和project文件,但是没有任务说明和hint还是很迷茫

**Reply by wozonet1** [reply]
刚看到有人做了wayback,那没事了

**Comment by Chih-EnShen** [comment]
课程不是闭源了，无法登陆

**Reply by Chih-EnShen** [reply]
没事了,刚刚看到wayback machine, 感谢各位大佬

**Comment by SeeingDragon** [comment]
su20的Wayback Machine网址:  https://web.archive.org/web/20200806162327/https://cs61c.org/su20/

**Reply by RunningNmXiao** [reply]
这个网址看不到slide，得用前面有人分享的日期新一点的。

**Reply by 123587** [reply]
有没有能看到project中图片的

**Comment by Leo-Lsc** [comment]
现在伯克利的网站都不开源了？

**Reply by zhangbird050801** [reply]
cs61B还可以看以前的，然后可以用 Wayback Machine 看以前的网站，不需要登录

**Reply by killus-tianyi** [reply]
wayback是不是也被ban 了？

**Comment by WangYuanze111** [comment]
伯克利的课程以后还有开放往年资源的可能吗？现在找资源都好麻烦。

**Comment by FFForeverCode** [comment]
现在不开源了怎么办

**Comment by Chih-EnShen** [comment]
花了一点时间从wayback machine把slides和其他一些materials下载下来了, 欢迎PR
https://github.com/Ch-EnShen/cs61c-all-materials

**Reply by mces89** [reply]
lab和project点不进去了？

**Reply by mces89** [reply]
fa20的

**Comment by jnulzl** [comment]
2017秋和2018秋的还今天(2024.07.09)可以打开：http://wla.berkeley.edu/~cs61c/fa17/和http://wla.berkeley.edu/~cs61c/fa18/

**Comment by linzhenpl07** [comment]
有没有人有fa20或者su0的lab和pro的任务说明？

**Reply by Jeandeng47** [reply]
可以用wayback machine输入网址然后查看

**Comment by tangzzycc** [comment]
有人有sp24的资源吗，没做完，现在看官网已经没了

**Comment by changwei-jing** [comment]
https://www.learncs.site/docs/curriculum-resource/cs61c/syllabus

**Reply by liujw23** [reply]
太感谢了

**Reply by followflowers** [reply]
mvp

**Reply by wuzhineihan** [reply]
mvp

**Reply by ZHHHH9980** [reply]
mvp

**Reply by MirRoR4s** [reply]
mvp

**Comment by q2333gh** [comment]
讲师 Dan Garcia 录制的视频 很有活力.超喜欢看!!!

**Comment by Chononut** [comment]
这个可以对标国内的组成原理吗

**Reply by WUADN** [reply]
可以,而且有部分OS预览

**Comment by airuikeplus** [comment]
最新课程网址：https://cs61c.org/fa24/

**Comment by Ericufo** [comment]
想请问下大家61C的homework和lab及project的区别有哪些呢，还有就是Berkeley往年是不是都没有开放homework呀

**Reply by trace1729** [reply]
问就是越新越好，对往年都没有开放 homework

**Comment by zhangbird050801** [comment]
RISC-V 部分也可以看看这个视频里面的讲解，讲的很清楚【[完结] 循序渐进，学习开发一个RISC-V上的操作系统 - 汪辰 - 2021春】 https://www.bilibili.com/video/BV1Q5411w7z5/?share_source=copy_web&vd_source=f5b55a19bbc981b8c6886f6d96d3b599

**Comment by 94jianaa** [comment]
现在2020fall需要伯克利账号才能进去，我用wayback machine找到了2020fall的lab和project等资源https://web.archive.org/web/20240423100311/https://inst.eecs.berkeley.edu/~cs61c/fa20/#projects

**Comment by InsideEmpire** [comment]
把[Fall2020](https://github.com/InsideEmpire/CS61C-Assignment)的作业原框架整理出来了

**Comment by RisingUppercut** [comment]
我做完了24fall的，除了project在本地做不了，其他都可以做。
24fall的官网能在archive上找到，目前全部资源都有: 
  https://web.archive.org/web/20241224171225/http://cs61c.org/fa24/  
recording 也可以在b站上看，b站有人上传了。

我的labs和projs实现（除了proj4）：
https://github.com/RisingUppercut/CS61C_2024_Fall

**Reply by RisingUppercut** [reply]
project4在本地做不了 ，少打了个4

**Reply by RisingUppercut** [reply]
archive的课程网页里面的所有谷歌云盘链接把前面的多余前缀删掉就能访问了

**Reply by Knight-of-Thunder** [reply]
现在是不是打不开了

**Reply by Knight-of-Thunder** [reply]
佬，https://web.archive.org/web/20241224171225/http://cs61c.org/fa24/这个链接好像打不开了

**Reply by RisingUppercut** [reply]
应该是web archive挂了，过一段时间再试试

**Reply by Knight-of-Thunder** [reply]
好呢，谢谢

**Comment by Knight-of-Thunder** [comment]
想问一下，使用24fa指定的logisim打开，为什么会出现逻辑门渲染（比如与门只显示一部分，不显示全部）异常的情况，我的java版本为jdk17。

**Reply by Knight-of-Thunder** [reply]
解决了，使用X11显示协议就可以了，wayhome对java的GUI兼容不好

**Comment by KDBRonaldo** [comment]
在中文和外国互联网上搜了好久都找不到这门课的homework，有大佬有吗呜呜呜。只做lab和project的话练习量足够吗？

**Reply by InsideEmpire** [reply]
Hi, homework 一般不对外开放，lab 和 project 作业量已经够了

**Comment by RisingUppercut** [comment]
我有22fall的hw


---- 回复的原邮件 ----
发件人  ***@***.***> 日期  2025年07月27日 10:10 收件人  ***@***.***> 抄送至  David ***@***.***>***@***.***> 主题  Re: [PKUFlyingPig/cs-self-learning] UCB CS61C: Great Ideas in Computer Architecture - CS自学指南 (Discussion #80)  

Hi, homework 一般不对外开放，lab 和 project 作业量已经够了

**Comment by RisingUppercut** [comment]
300202956这个q群里有不同学期的hw


---- 回复的原邮件 ----
发件人  ***@***.***> 日期  2025年07月27日 10:10 收件人  ***@***.***> 抄送至  David ***@***.***>***@***.***> 主题  Re: [PKUFlyingPig/cs-self-learning] UCB CS61C: Great Ideas in Computer Architecture - CS自学指南 (Discussion #80)  

Hi, homework 一般不对外开放，lab 和 project 作业量已经够了

**Reply by Mock-Canine** [reply]
您好，能麻烦您把群里的hw发到我的邮箱 2286596153@qq.com 吗，我申请进群但是管理员还没有通过，不知道群里现在还活不活跃了，谢谢。

**Reply by RisingUppercut** [reply]
发你了

**Reply by Heytrace007** [reply]
您好，能也麻烦你一下把群里的hw发到我的邮箱737752306@qq.com嘛,十分感谢。

**Reply by learn13-tech** [reply]
@RisingUppercut can  you please make a telegram group since the current one does not work, specifically for homework, thanks !

**Reply by hangocanh2303** [reply]
@RisingUppercut can you share homework for cs61c with me? please sent hw to my email: hangocanh2303@gmail.com
, thanks.

**Comment by csqh11** [comment]
官网能找到项目起始文件吗，我完全找不到

**Reply by csqh11** [reply]
我找到了，https://github.com/61c-teach/fa24-lab-starter

**Comment by ukeSJTU** [comment]
我参加了su25的CS61C课程，大家如果对课程内容有疑问可以回复本帖子，我们可以一起探讨～

**Reply by EternityXuNuo** [reply]
有课程视频的recording吗

**Reply by ukeSJTU** [reply]
我个人感觉可能是夏季学期的原因，8周课程里面有4周是Justin Yokota这个讲师讲的，质量很高，剩下的4周分别是4个不同的助教讲的，水平很一般，节奏控制的不行

**Comment by zxd369935044** [comment]
请问一下谁有su25的slides，今天发现突然进不去了

**Reply by ukeSJTU** [reply]
你可以把你邮箱给我，我可以打包发你

**Reply by zxd369935044** [reply]
OKOK，369935044@qq.com，谢谢谢谢！！

**Reply by ukeSJTU** [reply]
所有其他也需要资料的人请访问这个google-drive链接：https://drive.google.com/file/d/1GXEaGRhg3w4XKXgaf8Ms1fa5OjCTbNz7/view?usp=sharing

**Reply by LeoriumDev** [reply]
想問有 homework 嗎？

**Reply by ukeSJTU** [reply]
不好意思，作业是在 prairielearn 这个平台上进行的，我找不到任何下载的按钮

**Comment by csqh11** [comment]
吐槽一下这个logisim,他妈的画的时候线太容易缠住了，而且所有线都得自己拉过去，我画一个图的时候复制相同模组，拖动的时候不小心缠在一起了，还撤回不了，心态崩了

**Comment by csqh11** [comment]
我想问一下fa24的proj3是不是有些问题我测试的所有指令结果都比参考指令滞后一个周期，这是为啥

**Reply by csqh11** [reply]
牛鼻，同一个电路啥也没改，现在通过了，nice

**Reply by Zlatanwic** [reply]
想问一下同学你是在哪里测试的

**Comment by Zlatanwic** [comment]
请问一下cs61c的project本地测试就可以吗

**Comment by csqh11** [comment]
是的



发自我的iPhone


------------------ 原始邮件 ------------------
发件人: Rodebiau ***@***.***&gt;
发送时间: 2025年9月10日 14:25
收件人: PKUFlyingPig/cs-self-learning ***@***.***&gt;
抄送: csqh11 ***@***.***&gt;, Comment ***@***.***&gt;
主题: Re: [PKUFlyingPig/cs-self-learning] UCB CS61C: Great Ideas in Computer Architecture - CS自学指南 (Discussion #80)





 
请问一下cs61c的project本地测试就可以吗

**Comment by Bushiganyuanzuohuaizi** [comment]
我是一个CS小白，尽管大二，但代码能力差。我觉得这个网站以及这个课程很好，但也有难点。
我认为这里的课程难点有三:
课程节奏太快，对非英语母语的学生不友好，我于是没上课而是直接看PPT，但是看得还是很费劲。我在想要不要看其他的相同知识的中文课程，然后只写这里的题与项目。
项目的前置部分，我好像拉取不了它的仓库，那些命令行也不知道干嘛用的。我第一个项目敲完代码，结果在终端按照它的命令行的指令去做，根本运行不了，更别说调试了。好像又不能在IDE里面搞，必须用终端，而且还得是linux系统。哎难办额。
以及与学业的平衡，作为一个想保研但绩点不太够的学生，我要把大片的课余时间用在这上面，很可能是150小时甚至更多，我不知道应不应该。但是我们的课程很浅，也没多少内容，我总感觉作为一个计算机学生得自学一些技术栈
大佬们怎么办，我很迷茫，经常把精力放在犹豫与徘徊上

**Reply by Bushiganyuanzuohuaizi** [reply]
我经常大半天耗在这上面，总感觉收获寥寥，因为我时而纠结时而做事，只有在写项目的时候才稍微专注下来。我不知道我怎么这么容易陷入纠结当中，导致我为数不多的能量一点点被耗尽，每次我来学的时候都不太兴奋，甚至有些失落。经常是下定决心学这门课程却潦草收尾，或者效率极低。

**Reply by Lceoliu** [reply]
lab0主要是用来熟悉Berkeley的集群和git的，其实没有必要，后面的都可以在本地上完成；然后其实我感觉学到最多的是Homework部分，课程的话其实非母语还是容易走神或者听不懂的

**Comment by Lceoliu** [comment]
25fall 内容
https://cs61c.org/fa25/

**Comment by DayDaySpeed** [comment]
想省时间的可以直接写项目3，有详细注释和步骤  https://github.com/DayDaySpeed/CS61C-CPU2024.git

**Comment by cuckooaa** [comment]
61c的lecture讲课视频是看不了吗/(ㄒoㄒ)/~~怎么要登录账号

**Reply by RisingUppercut** [reply]
24fall的可以

**Comment by bitterlucky** [comment]
fa20年的floating point那个live lecture是不是没必要看，我看底下评论说:This live is mostly about exams, schedule and out-of-dated 'news'. There is little chance that you need to watch this if you only want to finish the course

**Comment by 54liujingcheng** [comment]
https://github.com/54liujingcheng/CS61C-my-project/tree/main
这是我学CS61C做的project，2020Fall版本

**Comment by CjtrusZ** [comment]
各位大佬，请问这门课程是不是包含了数字逻辑电路这门课程呀

**Comment by WUADN** [comment]
只是一笔带过重点，不能兼容数字电路

**Comment by WUADN** [comment]
国内的数字电路还是专门找国内数字电路课比较好

**Comment by Mock-Canine** [comment]
有大佬知道cmu18447的lab有开源吗？目前只找到了sp24的网站和slides

**Comment by 1010fzp0101** [comment]
有人一起组队吗， 我现在刚刚开始学， qq: 450339151
